### PR TITLE
fix post processor restore manifest to be case-insensitive

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs [hunk 1]
```diff
@@ -297,10 +289,11 @@ private List<ManifestItem> RestoreIncrementalManifestItems(Manifest manifest)
             }
         }
 
-        private static Dictionary<string, List<ManifestItem>> GroupBySourceRelativePath(IEnumerable<ManifestItem> items)
+        private static OSPlatformSensitiveDictionary<List<ManifestItem>> GroupBySourceRelativePath(IEnumerable<ManifestItem> items)
         {
-            return (from f in items
-                    group f by f.SourceRelativePath).ToDictionary(g => g.Key, g => g.ToList());
+            var pairs = (from f in items
+                         group f by f.SourceRelativePath).Select(g => new KeyValuePair<string, List<ManifestItem>>(g.Key, g.ToList()));
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs]: `group .. by ..` will use default comparer.

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs]: good catch!

---
In reply to: [120803064](https://github.com/dotnet/docfx/pull/1736#discussion_r120803064) [](ancestors = 120803064)

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**